crates/biome_deserialize_macros/src/deserializable_derive.rs

```diff
@@ -415,25 +437,49 @@ fn generate_deserializable_struct(
     } else {
         validator
     };
-    let unknown_key_handler = match data.unknown_fields {
-        UnknownFields::Warn | UnknownFields::Deny => {
-            let with_customseverity = if data.unknown_fields == UnknownFields::Warn {
-                quote! { .with_custom_severity(biome_diagnostics::Severity::Warning) }
-            } else {
-                quote! {}
-            };
-            quote! {
-                unknown_key => {
-                    const ALLOWED_KEYS: &[&str] = &[#(#allowed_keys),*];
-                    diagnostics.push(DeserializationDiagnostic::new_unknown_key(
-                        unknown_key,
-                        key.range(),
-                        ALLOWED_KEYS,
-                    )#with_customseverity)
+    let unknown_key_handler = if data.rest_field.is_some() {
+        quote! {
+            unknown_key => {
+                let key_text = Text::deserialize(&key, "", diagnostics)?;
+                let value = Deserializable::deserialize(&value, key_text.text(), diagnostics)?;
+                unknown_fields.push((key_text, value));
```

Comment:
To avoid the allocation of an intermediary structure (here a `Vec`), we could assume that the type of the rest field implements [Extend<K, V>](https://doc.rust-lang.org/std/iter/trait.Extend.html). This is [the case of HashMap](https://doc.rust-lang.org/std/iter/trait.Extend.html#impl-Extend%3C(K,+V)%3E-for-HashMap%3CK,+V,+S%3E).

Then we could just write something like:

```
result.#rest_field.extend([(key_text, value)]);
```